tools/codegen/model_generator/model.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model_generator
+
+import (
+	"fmt"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	tmplExtns "istio.io/mixer/tools/codegen/template_extension"
+	"github.com/golang/protobuf/proto"
+)
+
+/*
+Data model to be used to generate code for istio/mixer
+*/
+type Model struct {
+	// top level fields
+	Name        string
+	Check       bool
```

Comment:
I'd prefer to use a method enum over a bool "check or not check". We're in the process of defining aspects whose lifecycle is a bit different (ingestion adapters) so I worry that structuring this code as "check or not check" will make it harder to change later. [Also, makes me think of this great scene.](https://www.youtube.com/watch?v=ACmydtFDTGs&t=10)